src/psyclone/tests/dependency_tools_test.py

```diff
@@ -0,0 +1,289 @@
+        dep_tools.get_all_messages()[0]
+
+    # Use a stencil access (with write), which prevents parallelisation
+    parallel = dep_tools.can_loop_be_parallelised(loops[3], "jj")
+    assert not parallel
+    assert "Variable mask is using index jj + 1 and jj and can therefore "\
+           "not be parallelised" in dep_tools.get_all_messages()[0]
+
+
+# -----------------------------------------------------------------------------
+def test_scalar_parallelise(parser):
+    '''Tests the scalar checks of can_loop_be_parallelised.
+    '''
+    reader = FortranStringReader('''program test
+                                 do jj = 1, jpj   ! loop 0
+                                    do ji = 1, jpi
+                                       a(ji, jj) = b
+                                     end do
+                                 end do
+                                 do jj = 1, jpj   ! loop 1
+                                    do ji = 1, jpi
+                                       b = a(ji, jj)
+                                     end do
+                                 end do
+                                 do jj = 1, jpj   ! loop 2
+                                    do ji = 1, jpi
+                                       b = a(ji, jj)
+                                       c(ji, jj) = b*b
+                                     end do
+                                 end do
+                                 do jj = 1, jpj   ! loop 3
+                                    do ji = 1, jpi
+                                       b = b + a(ji, jj)
+                                     end do
+                                 end do
+                                 end program test''')
+    prog = parser(reader)
+    psy = PSyFactory("nemo", distributed_memory=False).create(prog)
+    loops = psy.invokes.get("test").schedule
+    dep_tools = DependencyTools(["levels", "lat"])
+
+    # Read only scalar variable: a(ji, jj) = b
+    parallel = dep_tools.can_loop_be_parallelised(loops[0], "jj")
+    assert parallel
+
+    # Write only scalar variable: a(ji, jj) = b
+    parallel = dep_tools.can_loop_be_parallelised(loops[1], "jj")
+    assert not parallel
+    assert "Variable 'b' is only written once" \
+        in dep_tools.get_all_messages()[0]
+
+    # Write to scalar variable happens first
+    parallel = dep_tools.can_loop_be_parallelised(loops[2], "jj")
+    assert parallel
+
+    # Reduction operation on scalar variable
+    parallel = dep_tools.can_loop_be_parallelised(loops[3], "jj")
+    assert not parallel
+    assert "Variable 'b' is read first, which indicates a reduction."\
+        in dep_tools.get_all_messages()[0]
+
+
+# -----------------------------------------------------------------------------
+def test_derived_type(parser):
+    '''Tests assignment to derived type variables. For now (see #363)
+    they are only partially supported, and as default assume that these
+    kind of statements are parallelisable.
```

Comment:
Should that say "are not parallelisable"?

Reply:
Yes (I think I changed my mind long after having written the docs - originally I excepted derived types to be parallelisable, then I added the ability to explicitly disable this for given variables, which is much nicer, and changed the default, without changing the docs).